src/components/notificationMessage/NotificationMessage.jsx

```diff
@@ -1,13 +1,47 @@
 import React, { useState } from "react";
 import { IoMailUnreadSharp, IoMailOpenSharp } from "react-icons/io5";
 import "./notificationMessage.css";
+import { updateNotification } from "../../redux/notificationsSlice";
+import { useDispatch } from "react-redux";
 
 const NotificationMessage = ({ message, key }) => {
   const [showMessage, setShowMessage] = useState(false);
+  const [readMessage, setReadMessage] = useState(false);
+  const dispatch = useDispatch();
+
   const openMessage = () => {
     setShowMessage(!showMessage);
-    message.opened = true;
+    setReadMessage(true);
+    if (message.isOpened == false) {
+      try {
+        dispatch(
+          updateNotification({
+            ...message,
+            isOpened: true,
+          })
+        ).then((result) => {
+          setShowMessage(!showMessage);
+        });
+      } catch (error) {
+        console.error("Error opening message:", error);
+      }
+    }
   };
+
+  var date = new Date(message.createdAt);
+
+  var year = date.getFullYear();
+  var month = date.getMonth() + 1;
+  var day = date.getDate();
+
+  if (month < 10) {
+    month = "0" + month;
+  }
+  if (day < 10) {
+    day = "0" + day;
+  }
+  var formattedDate = year + "-" + month + "-" + day;
```

Comment:
imas na netu biblioteke koje lepo formatiraju datum kako god hoces, skloni ovu logiku i nadji to, msm da date-fns moze da ti pomogne